### DISCUSSION TITLE
[Feature] Disclaimer/Footer base64

### DISCUSSION THREAD

**Discussion by mgarbin** [rspamd/rspamd#3860] [Ideas]
Hi, 
on old sw altermime there was the possibility to add a disclaimer in base64 : 

        --disclaimer=, Set the plaintext disclaimer source file.
        --disclaimer-html=, Set the HTML disclaimer source file.
        --disclaimer-b64=, Set the BASE64 encoded disclaimer source file (implies --force-into-b64)

With rspamd we can add a disclaimer in text or html.
Adding base64 is useful for who send an email with a MUA that generate email in base64 .

**Comment by vstakhov** [comment]
It is not needed, as Rspamd will use (correctly) quoted-printable encoding for each text part, even those that were originally encoded with base64. `altermime` approach is totally wrong by design.

**Reply by trantor** [reply]
Mmm @vstakhov hopefully I won't be saying something terribly stupid, but let's assume for a second that one writes an e-mail, for instance, in a logographic script using UTF-8 as text encoding. That would mean quite a lot of characters encoded as 2 or 3-byte sequences. Let's also assume that their software prepares the message as acceptable on a 7-bit clean channel, hence passing the text through QP or Base64 encoding. Afaik QP-encoding would incur into a far heavier overhead  than Base64-encoding when it comes to increase in message size here.
So, wouldn't converting to QP forcefully be detrimental in such a situation (which I personally don't believe to be that far fetched)?
When you talk about the choice of QP as "correct", what do you mean, specifically?
Quoting from RFC 2045:
> The Quoted-Printable encoding is intended to represent data that
   largely consists of octets that correspond to printable characters in
   the US-ASCII character set.  It encodes the data in such a way that
   the resulting octets are unlikely to be modified by mail transport.
   If the data being encoded are mostly US-ASCII text, the encoded form
   of the data remains largely recognizable by humans.

In the case, for instance, of text in a non-latin script, the use of QP would not appear to fit the intended purpose well.
Could you provide some additional information on the subject?

Thanks in advance
